open-api/rest-catalog-open-api.yaml

```diff
@@ -3642,6 +3781,173 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+
+    PlanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+        stats-fields:
+          description:
+            A list of fields that the client requests the server to send statistics
+            in each `FileScanTask` returned in the response
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+
+    TableScanContext:
```

Comment:
I ended up using this to separate snapshot scan vs incremental scan, it feels like the cleanest to me compared to mixing the properties of these 2 different scans. Let me know what everyone thinks about this design.

Reply:
I think this is cleaner approach then duplicating the similar properties in the different type of scans.

Reply:
Having the type `TableScanContext` defined helps because you can use `oneOf` to select the properties defined by `TableScanContext` or send a `plan-task`. It also can be reused for the `PreplanTableRequest`.

I think in both cases, there does not need to be an extra object layer nested in `table-scan-context`. These properties should be defined at the top level like in the original version @rahil-c posted.